packages/world/src/systems/libraries/MoveLib.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { VoxelCoord, VoxelCoordLib } from "../../VoxelCoord.sol";
+
+import { ObjectTypeSchema, ObjectTypeSchemaData } from "../../codegen/tables/ObjectTypeSchema.sol";
+import { ObjectTypeMetadata } from "../../codegen/tables/ObjectTypeMetadata.sol";
+import { ObjectType } from "../../codegen/tables/ObjectType.sol";
+import { Position } from "../../codegen/tables/Position.sol";
+import { ReversePosition } from "../../codegen/tables/ReversePosition.sol";
+import { PlayerPosition } from "../../codegen/tables/PlayerPosition.sol";
+import { ReversePlayerPosition } from "../../codegen/tables/ReversePlayerPosition.sol";
+import { ActionType } from "../../codegen/common.sol";
+import { Energy, EnergyData } from "../../codegen/tables/Energy.sol";
+
+import { ObjectTypeId, AirObjectID, PlayerObjectID } from "../../ObjectTypeIds.sol";
+import { gravityApplies, inWorldBorder } from "../../Utils.sol";
+import { PLAYER_MOVE_ENERGY_COST } from "../../Constants.sol";
+import { notify, MoveNotifData } from "../../utils/NotifUtils.sol";
+import { TerrainLib } from "./TerrainLib.sol";
+import { EntityId } from "../../EntityId.sol";
+import { transferEnergyFromPlayerToPool } from "../../utils/EnergyUtils.sol";
+import { requireValidPlayer } from "../../utils/PlayerUtils.sol";
+
+library MoveLib {
```

Comment:
Moved `MoveLib` back into its own file as its used in multiple systems. Consolidate gravity lib function into `MoveLib` otherwise causes circular dependency